blurring_as_a_service/metrics/custom_metrics_calculator.py

```diff
@@ -0,0 +1,500 @@
+                            category_stats["false_negatives_large"] += 1
+
+    def _get_gt_boxes_and_predicted_boxes_per_category(
+        self, tagged_validation_folder: str
+    ):
+        """
+        For each category of each file in tagged_validation_folder it gathers the ground truth boxes and all
+         predicted boxes. This will be later used in computing the total blurred area per category.
+        Args:
+            tagged_validation_folder: Path to the folder containing all the tagged validation values.
+
+        Returns:
+
+        """
+        for tagged_validation_file in self.get_all_filenames_in_dir(
+            tagged_validation_folder
+        ):
+            with open(f"{tagged_validation_folder}/{tagged_validation_file}") as f:
+                tagged_validation_content = json.load(f)
+                if any(
+                    len(tagged_validation_content[key])
+                    != len(tagged_validation_content["GT_labels"])
+                    for key in ["GT_boxes", "TP_labels"]
+                ):
+                    raise Exception(
+                        f"{tagged_validation_file} not well formed, "
+                        f"the len of GT_boxes, GT_labels and TP_labels is not matching."
+                    )
+
+                for i, (gt_box, gt_label) in enumerate(
+                    zip(
+                        tagged_validation_content["GT_boxes"],
+                        tagged_validation_content["GT_labels"],
+                    )
+                ):
+                    category_stats = self._statistics_per_category[gt_label]
+                    category_stats["gt_boxes"].append(
+                        [tagged_validation_content["GT_boxes"][i]]
+                    )
+                    category_stats["pred_boxes"].append(
+                        tagged_validation_content["Pred_boxes"]
+                    )
+
+    def _get_and_prepare_categories(self, coco_file_with_categories):
+        """
+        Retrieves all the possible categories and values from the coco file and creates the _statistics_per_category
+        attribute that is used to generate the metrics.
+
+        Parameters
+        ----------
+        coco_file_with_categories
+
+        Returns
+        -------
+
+        """
+        with open(coco_file_with_categories) as f:
+            categories_file_content = json.load(f)
+        self._bias_category_mapper = BiasCategoryMapper(
+            categories_file_content["categories"]
+        )
+        all_grouped_categories = self._bias_category_mapper.get_all_grouped_categories()
+
+        self._statistics_per_category = {}
+        for d in all_grouped_categories:
+            for attr_1, attr_2, attr_3 in [
+                d["name"].split("/") + [""] * (3 - len(d["name"].split("/")))
+            ]:
+                if attr_1 == "licence_plate":
+                    self._statistics_per_category[d["id"]] = {
+                        "id": d["id"],
```

Comment:
Can we add in the if-else only the ones that are not in common and before or after add the ones are in common so we don't repeat them?